😎 - Refactor TimeManipulationProvider

> 🌻 This issue is perfect for developers who are new to Miunie!

## Summary

The [TimeManipulationProvider](https://github.com/control-net/Miunie/blob/master/src/Miunie.Core/Providers/TimeManipulationProvider.cs) is able to convert inputs such as `20 hours` into a `TimeSpan` of that value.

The current implementation is using a large if/else if/else if/... statement.
This would ideally become a switch statement. In fact, this would be an ideal use for [switch expressions](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#switch-expressions) that were introduced in C# 8. :blush: 

## What to look out for

We currently have a suite of unit tests for this class. Make sure you run the unit tests either from visual studio or by running the following from a terminal:

```
dotnet test src/Miunie.Core.XUnit.Tests
```

## Interested? 😊 

If you'd like to help us by solving this issue, just say so in the comments. ❤️ 
And don't shy away from asking any kind of question you might have.

Should you have any questions at all, feel free to contact either me (@petrspelos) or @DraxCodes on our Discord server:

[![Discord](https://img.shields.io/discord/377879473158356992?color=7289DA&label=Discord%20Server&style=for-the-badge)](https://discord.gg/cGhEZuk)

_And of course, thank you so much for contributing! 🎉 It's because of amazing people like yourself that Miunie exists._

I've written a simple switch expression-
```c#        
 return timeframe switch {
                var tframe when (
                    tframe == "hours" || tframe == "hour" || tframe == "hrs" || tframe == "hr"
                    ) => new TimeSpan(units, 0, 0),
                var tframe when (
                    tframe == "minutes" || tframe == "minute" || tframe == "mins" || tframe == "min"
                    ) => new TimeSpan(units, 0, 0),
                var tframe when (
                    tframe == "seconds" || tframe == "second" || tframe == "secs" || tframe == "sec"
                    ) => new TimeSpan(units, 0, 0),  
                _ => null,
            };
```
But I get this error when I try to use a switch expression -

> Providers/TimeManipulationProvider.cs(32,20): error CS8370: Feature 'recursive patterns' is not available in C# 7.3. Please use language version 8.0 or greater.

It says it doesn't support switch expression for the current language version of this project. How should this issue be resolved. Shall I fall back to writing switch statement, or is there a way around to get this done with switch expression?

--- a/src/Miunie.Core/Miunie.Core.csproj
+++ b/src/Miunie.Core/Miunie.Core.csproj
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
     <CodeAnalysisRuleSet>../.ruleset</CodeAnalysisRuleSet>
+    <LangVersion>8.0</LangVersion>
   </PropertyGroup>
 
   <ItemGroup>

--- a/src/Miunie.Core/Providers/TimeManipulationProvider.cs
+++ b/src/Miunie.Core/Providers/TimeManipulationProvider.cs
@@ -29,22 +29,18 @@ namespace Miunie.Core.Providers
 
             timeframe = timeframe.Trim().ToLower();
 
-            if (timeframe == "hours" || timeframe == "hour" || timeframe == "hrs" || timeframe == "hr")
-            {
-                return new TimeSpan(units, 0, 0);
-            }
-            else if (timeframe == "minutes" || timeframe == "minute" || timeframe == "mins" || timeframe == "min")
-            {
-                return new TimeSpan(0, units, 0);
-            }
-            else if (timeframe == "seconds" || timeframe == "second" || timeframe == "secs" || timeframe == "sec")
-            {
-                return new TimeSpan(0, 0, units);
-            }
-            else
-            {
-                return null;
-            }
+            return timeframe switch {
+                var tframe when
+                    tframe == "hours" || tframe == "hour" || tframe == "hrs" || tframe == "hr"
+                    => new TimeSpan(units, 0, 0),
+                var tframe when
+                    tframe == "minutes" || tframe == "minute" || tframe == "mins" || tframe == "min"
+                    => new TimeSpan(0, units, 0),
+                var tframe when
+                    tframe == "seconds" || tframe == "second" || tframe == "secs" || tframe == "sec"
+                    => new TimeSpan(0, 0, units),
+                _ => null,
+            };
         }
 
         public DateTime? GetDateTimeLocalToUser(DateTime? utcDateTime, MiunieUser user)

